test/solidity/reentrancy/reentrancy.js

```diff
@@ -0,0 +1,74 @@
+/*-
+ *
+ * Hedera Smart Contracts
+ *
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+const { expect } = require('chai')
+const { ethers } = require('hardhat')
+const Constants = require('../../constants')
+
+describe('Reentrancy Guard', function () {
+  const tenHBAR = ethers.utils.parseEther("10.0");
+  async function deployContractsAndSendHbars() {
+    const [owner] = await ethers.getSigners();
+    const factorySender = await ethers.getContractFactory(
+        Constants.Contract.ReentrancyGuardTestSender
+    )
+    contractSender = await factorySender.deploy()
+    const factoryReceiver = await ethers.getContractFactory(
+        Constants.Contract.ReentrancyGuardTestReceiver
+      )
+    contractReceiver = await factoryReceiver.deploy(contractSender.address)
+    const transactionHash = await owner.sendTransaction({
```

Comment:
nit: we can add a payable constructor to `ReentrancyGuardTestSender`, fund it on contract deployment, and remove that tx completely